src/yunohost/app.py

```diff
@@ -829,12 +829,110 @@ def app_makedefault(auth, app, domain=None):
         raise MoulinetteError(errno.EPERM,
                               m18n.n('ssowat_persistent_conf_write_error', error=e.strerror))
 
+    os.system('chmod 644 /etc/ssowat/conf.json.persistent')
+
+    logger.success(m18n.n('ssowat_conf_updated'))
+
+
+def app_removedefault(auth, app, domain=None):
+    """
+    Remove the default redirection to an app on a domain
+
+    Keyword argument:
+        app
+        domain
+
+    """
+    from yunohost.domain import domain_list
+
+    app_settings = _get_app_settings(app)
+    app_domain = app_settings['domain']
```

Comment:
You don't have any guarantee that [`domain` will be present in `app_settings`](https://github.com/likeitneverwentaway/yunohost/blob/252928717d2c0210fa00ac13f769e9be48333362/src/yunohost/app.py#L303), you need to do a `app_settings.get('domain')` instead and return if this app doesn't have a domain (some does).